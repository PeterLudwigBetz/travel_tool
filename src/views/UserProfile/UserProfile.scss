form.new-profile {
    fieldset {
      .input-group {
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
      }
    }
}

h1{
    height: 23px;
    width: 72px;
    color: #4F4F4F;
    font-family: "DIN Pro";
    font-size: 18px;
    font-weight: 800;
    line-height: 23px;
    margin-left: 0px;
  }
  .main{
    height: 400px;
    width: 1080px;
    border: 1px solid #E4E4E4;
    border-radius: 2px;
    background-color: #FFFFFF;
    margin-left: 0px;
    margin-top: 10px;
    margin-right: 60px;
    padding-bottom: 50px;
  }
  .main--user_profile{
    width:550px;
    padding: 0px 0px 10px 20px;
  }
.container--handleSubmit{
    margin-right:320px;
    height: 23px;
    width: 72px;
    color: #4F4F4F;
    font-family: "DIN Pro";
    font-size: 18px;
    font-weight: 500;
    line-height: 23px;
}
.profile-divider{
    box-sizing: border-box;
    height: 2px;
    width: 980px;
    border: 1px solid #E4E4E4;
}
.main--user_profile{
    padding: 30px 0px 60px 50px;
    height: 500px;
}
