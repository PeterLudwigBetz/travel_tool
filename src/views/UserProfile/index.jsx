import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { FormContext } from '../../components/Forms/FormsAPI';
import PersonalDetailsFieldset from '../../components/Forms/NewRequestForm/FormFieldsets/PersonalDetails';
import SubmitArea from '../../components/Forms/NewRequestForm/FormFieldsets/SubmitArea';
import '../../components/Forms/NewRequestForm/NewRequestForm.scss';
import './UserProfile.scss';
import updateUserProfile from '../../redux/actionCreator/userProfileActions';
import ProfileForm from '../../components/Forms/ProfileForm';
import Base from '../Base';
import { fetchRoleUsers } from '../../redux/actionCreator/roleActions';

class UserProfile extends Base {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const { fetchRoleUsers } = this.props;
    fetchRoleUsers(53019);
  }

  render() {
    const { manager, updateUserProfile, user, fetchUserData, getUserData } = this.props;
    return (
      <Fragment>
        <h1>PROFILE</h1>
        <div className="main">
          <div className="main--user_profile">
            <ProfileForm
              managers={manager}
              updateUserProfile={updateUserProfile}
              userData={fetchUserData && fetchUserData.result}
              user={user}
              getUserData={getUserData}
            />
          </div>
        </div>
      </Fragment>
    );
  }
}

UserProfile.defaultProps = {
  manager: []
};

export const mapStateToProps = ({ user, role }) => ({
  ...role,
  fetchUserData: user.getUserData
});

const actionCreators = {
  updateUserProfile,
  fetchRoleUsers,
};
export default connect(
  mapStateToProps,
  actionCreators
)(UserProfile);
