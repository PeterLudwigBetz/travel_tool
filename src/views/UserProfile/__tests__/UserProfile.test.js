import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import UserProfile from '../index';

const props = {
  user: {
    UserInfo: {
      name: 'John Doe',
      id: '-LHJlG',
      picture: 'http://www.image.com/jepg'
    }
  },
  isLoading: false,
  history: {
    push: jest.fn()
  },
  errors: [],
  shouldOpen: false,
  onNotificationToggle: jest.fn(),
};

const initialState = {
  auth: {
    user: {
      UserInfo: {
        name: 'John Doe',
        id: '-LHJlG',
        picture: 'http://www.image.com/jepg'
      }
    }
  },
  errors: [],
};

const mockStore = configureStore();
const store = mockStore(initialState);

describe('<UserProfile />', ()=>{
  it('should render correctly', () =>{
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <UserProfile {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
