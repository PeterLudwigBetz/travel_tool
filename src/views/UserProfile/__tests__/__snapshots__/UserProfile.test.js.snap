// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> should render correctly 1`] = `
Array [
  <h1>
    PROFILE
</h1>,
  <div
    className="main"
>
    <div
        className="main--user_profile"
    >
        <form
            className="new-profile"
            onSubmit={[Function]}
        >
            <fieldset
                className="personal-details"
            >
                <legend
                    style={
                        Object {
                            "width": "100%",
                          }
                    }
                >
                    <span
                        className="personal-details-text"
                    >
                        Personal Details
                    </span>
                    <span
                        className="required-field"
                    >
                        * Required Field
                    </span>
                    <div
                        className="profile-divider"
                    />
                </legend>
                <div
                    className="input-group profile-input"
                >
                    <div
                        className="form-input  "
                    >
                        <label
                            htmlFor="name"
                        >
                            Name
                            <span
                                style={
                                    Object {
                                        "color": "red",
                                      }
                                }
                            >
                                *
                            </span>
                            <span>
                                Â 
                                (As it appears on your passport)
                            </span>
                        </label>
                        <input
                            className=" "
                            label="Name"
                            name="name"
                            onBlur={[Function]}
                            onChange={[Function]}
                            type="text"
                            value=""
                        />
                        <span
                            className="error"
                        />
                    </div>
                    <div
                        className="form-input  "
                    >
                        <label
                            htmlFor="gender"
                        >
                            Gender
                            <span
                                style={
                                    Object {
                                        "color": "red",
                                      }
                                }
                            >
                                *
                            </span>
                        </label>
                        <div
                            className="input "
                        >
                            <button
                                className="bg-btn bg-btn--inactive"
                                data-value="Female"
                                key="Female"
                                name="gender"
                                onClick={[Function]}
                                type="button"
                            >
                                Female
                            </button>
                            <button
                                className="bg-btn bg-btn--inactive"
                                data-value="Male"
                                key="Male"
                                name="gender"
                                onClick={[Function]}
                                type="button"
                            >
                                Male
                            </button>
                        </div>
                        <span
                            className="error"
                        />
                    </div>
                    <div
                        className="form-input  "
                    >
                        <label
                            htmlFor="role"
                        >
                            Role
                            <span
                                style={
                                    Object {
                                        "color": "red",
                                      }
                                }
                            >
                                *
                            </span>
                        </label>
                        <div
                            style={
                                Object {
                                    "position": "relative",
                                  }
                            }
                        >
                            <div
                                choices={
                                    Array [
                                        "Software Developer",
                                        "Technical Team Lead",
                                        "Product Engineer",
                                        "Product Coordinator",
                                        "Travel Coordinator",
                                        "Country Director",
                                      ]
                                }
                                className="input select  "
                                error=""
                                label="Role"
                                name="role"
                                onChange={[Function]}
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                role="button"
                                tabIndex="0"
                                type="dropdown-select"
                                value=""
                            >
                                <div
                                    className="value"
                                    style={
                                        Object {
                                            "width": undefined,
                                          }
                                    }
                                >
                                    <div
                                        className="select-dropdown inactive"
                                    >
                                        <img
                                            alt="icn"
                                            src="form_select_dropdown.svg"
                                        />
                                    </div>
                                </div>
                                <ul
                                    className="select-menu select-menu--inactive"
                                >
                                    <li
                                        key="Software Developer"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Software Developer
                                        </div>
                                    </li>
                                    <li
                                        key="Technical Team Lead"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Technical Team Lead
                                        </div>
                                    </li>
                                    <li
                                        key="Product Engineer"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Product Engineer
                                        </div>
                                    </li>
                                    <li
                                        key="Product Coordinator"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Product Coordinator
                                        </div>
                                    </li>
                                    <li
                                        key="Travel Coordinator"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Travel Coordinator
                                        </div>
                                    </li>
                                    <li
                                        key="Country Director"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Country Director
                                        </div>
                                    </li>
                                </ul>
                            </div>
                        </div>
                        <span
                            className="error"
                        />
                    </div>
                    <div
                        className="form-input  "
                    >
                        <label
                            htmlFor="department"
                        >
                            Department
                            <span
                                style={
                                    Object {
                                        "color": "red",
                                      }
                                }
                            >
                                *
                            </span>
                        </label>
                        <div
                            style={
                                Object {
                                    "position": "relative",
                                  }
                            }
                        >
                            <div
                                choices={
                                    Array [
                                        "Talent & Development",
                                        "Success",
                                        "IT",
                                        "Operations",
                                      ]
                                }
                                className="input select  "
                                error=""
                                label="Department"
                                name="department"
                                onChange={[Function]}
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                role="button"
                                tabIndex="0"
                                type="dropdown-select"
                                value=""
                            >
                                <div
                                    className="value"
                                    style={
                                        Object {
                                            "width": undefined,
                                          }
                                    }
                                >
                                    <div
                                        className="select-dropdown inactive"
                                    >
                                        <img
                                            alt="icn"
                                            src="form_select_dropdown.svg"
                                        />
                                    </div>
                                </div>
                                <ul
                                    className="select-menu select-menu--inactive"
                                >
                                    <li
                                        key="Talent & Development"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Talent & Development
                                        </div>
                                    </li>
                                    <li
                                        key="Success"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Success
                                        </div>
                                    </li>
                                    <li
                                        key="IT"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            IT
                                        </div>
                                    </li>
                                    <li
                                        key="Operations"
                                    >
                                        <div
                                            onClick={[Function]}
                                            onKeyPress={[Function]}
                                            role="button"
                                            tabIndex="0"
                                        >
                                            Operations
                                        </div>
                                    </li>
                                </ul>
                            </div>
                        </div>
                        <span
                            className="error"
                        />
                    </div>
                    <div
                        className="form-input  "
                    >
                        <label
                            htmlFor="manager"
                        >
                            Line Manager
                            <span
                                style={
                                    Object {
                                        "color": "red",
                                      }
                                }
                            >
                                *
                            </span>
                            <span>
                                Â 
                                (Manager)
                            </span>
                        </label>
                        <div
                            style={
                                Object {
                                    "position": "relative",
                                  }
                            }
                        >
                            <div
                                choices={Array []}
                                className="input select  "
                                error=""
                                label="Line Manager"
                                name="manager"
                                onChange={[Function]}
                                onClick={[Function]}
                                onKeyPress={[Function]}
                                role="button"
                                tabIndex="0"
                                type="dropdown-select"
                                value=""
                            >
                                <div
                                    className="value"
                                    style={
                                        Object {
                                            "width": undefined,
                                          }
                                    }
                                >
                                    <div
                                        className="select-dropdown inactive"
                                    >
                                        <img
                                            alt="icn"
                                            src="form_select_dropdown.svg"
                                        />
                                    </div>
                                </div>
                                <ul
                                    className="select-menu select-menu--inactive"
                                />
                            </div>
                        </div>
                        <span
                            className="error"
                        />
                    </div>
                </div>
            </fieldset>
            <div
                className="submit-area"
            >
                <button
                    className="bg-btn bg-btn--inactive"
                    disabled={true}
                    type="submit"
                >
                    Save Changes
                </button>
            </div>
        </form>
    </div>
</div>,
]
`;
